tensorflow/python/framework/python_op_gen.cc

```diff
@@ -310,16 +348,43 @@ string GenEagerPythonOp::Code() {
     param_names_.push_back(param_and_default.first);
   }
 
+  std::unordered_map<string, string> type_annotations;
+  // Only populate map for whitelisted ops
+  if (type_annotate_ops.find(op_def_.name()) != type_annotate_ops.end()) {
+    type_annotations = GetTypeAnnotationMap();
+  }
+
   string parameters;
   for (const auto& param : params_no_default_) {
     if (!parameters.empty()) strings::StrAppend(&parameters, ", ");
     strings::StrAppend(&parameters, param.GetRenameTo());
+
+    // Add type annotations to param
+    if (type_annotations.find(param.GetName()) != type_annotations.end()) {
+      if (!type_annotations[param.GetName()].empty()) {
+        strings::StrAppend(&parameters, ": ", type_annotations[param.GetName()]);
+      }
+    }
   }
+
   string parameters_with_defaults = parameters;
   for (const auto& param_and_default : params_with_default_) {
     if (!parameters.empty()) strings::StrAppend(&parameters, ", ");
     if (!parameters_with_defaults.empty())
       strings::StrAppend(&parameters_with_defaults, ", ");
+
+    // Add type annotations to param_and_default
+    if (type_annotations.find(param_and_default.first.GetName()) != type_annotations.end()) {
+      if (!type_annotations[param_and_default.first.GetName()].empty()) {
```

Comment:
Suggest using a param_name local to avoid repeated calls - it makes the code cleaner, too.